ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala

```diff
@@ -169,15 +178,39 @@ object LedgerContext {
 
     require(esf != null, "ExecutionSequencerFactory must not be null.")
 
-    def ledgerId: String =
+    var cachedLedgerId: Option[domain.LedgerId] = None
+
+    def ledgerId: domain.LedgerId =
       configuredLedgerId match {
-        case LedgerIdMode.Static(id) => id.unwrap
+        case LedgerIdMode.Static(id) => id
         case LedgerIdMode.Dynamic() =>
-          LedgerIdentityServiceGrpc
-            .blockingStub(channel)
-            .getLedgerIdentity(GetLedgerIdentityRequest())
-            .ledgerId
+          cachedLedgerId.synchronized {
```

Comment:
wrong concurrency semantics. I mentioned this already in #1400. Why do you have it in this PR too?

Reply:
One PR depends on another, so one is the subset of another. I will send them as non-draft PRs one by one.